mantidproject/mantid
# Improving ExtractFFTSpectrum Algorithm

Made the following improvements to ExtractFFTSpectrum
1. Added properties `Shift`, `Autoshift`, and `AcceptXRoundingErrors`. These are properties of FFT that were apparently missing according to the reporter of the issue.
2. On only real inputs, ExtractFFTSpectrum left a 'tail' of zeros on each histogram, due to the way FFT deals with this. These now get chopped off such that no real data is lost.
3. ExtractFFTSpectrum now returns the units decided by FFT. 



**To test:**


_New Properties_
- For the properties, ensure that enabling each new property gives the correct outputs (I'm not too sure how this is supposed to look vs regular FFT)

_Tail zeros_
1. Load `IRS26176_ipg.nxs` dataset and rebin it with the params `-0.5,0.005,0.5`.
2. Run ExtractFFTSpectrum, ensuring the `InputImagWorkspace` property is empty.
3. Show the data, there should be 100 X values. 
4. Now run ExtractFFTSpectrum again, this time setting `InputImagWorkspace` to be the same as `InputWorkspace`
5. Show the data, there should be 200 X values

_Units_
1. Load any dataset and rebin to ensure all bins are of equal width.
2. Run both ExtractFFTSpectrum and FFT.
3. Ensure the units produced by ExtractFFTSpectrum match those of FFT.

Fixes #26995 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
